functions-methods.js

```diff
@@ -0,0 +1,68 @@
+// Je gaat functies schrijven die we kunnen hergebruiken om sommige emailadressen te checken. Nu zul je gaan merken hoe handig functies kunnen zijn!
+// Je zult hier methoden van het String Object voor nodig hebben, dus pak de paragraaf op EdHub over het String Object er even bij.
+
+
+/* Opdracht  1 */
+// Schrijf een functie genaamd getEmailDomain, die een emailadres verwacht en de domeinnaam teruggeeft. Een domeinnaam is hetgeen dat na het @ in het adres staat
+// ---- Verwachte uitkomsten:
+// getEmailDomain("n.eeken@novi-education.nl") geeft novi-education.nl
+// getEmailDomain("t.mellink@novi.nl") geeft novi.nl
+// getEmailDomain("a.wiersma@outlook.com") geeft outlook.com
+
+function getEmailDomain (input){
+    return input.substring(input.indexOf("@") + 1)
+}
+
+const emailDomain = getEmailDomain("n.eeken@novi-education.nl")
+console.log(emailDomain)
+
+
+/* Opdracht  2 */
+// Schrijf een functie genaamd typeOfEmail, die een emailadres verwacht. De functie checkt of het emailadres een novi domein heeft (medewerker), een novi-education domein (student), of extern domein (zoals gmail of outlook)
+// ---- Verwachte uitkomsten:
+// typeOfEmail("n.eeken@novi-education.nl") geeft "Student"
+// typeOfEmail("t.mellink@novi.nl") geeft geeft "Medewerker"
+// typeOfEmail("novi.nlaapjesk@outlook.com") geeft geeft "Extern" <-- deze moet het ook doen!
+// typeOfEmail("a.wiersma@outlook.com") geeft "Extern"
+
+function typeOfEmail (input){
+    if (input.includes("@novi-education")){
+        return "Student"
+    } else if (input.includes("@novi.nl")){
+        return "Medewerker"
+    } else {
+        return "Extern"
+    }
+}
+
+const email = typeOfEmail("novi.nlaapjesk@outlook.com")
+console.log(email)
+
+/* Opdracht  3 */
+// Schrijf een functie genaamd checkEmailValidity, die een emailadres verwacht en checkt of het emailadres valide is. De functie returned true of false, afhankelijk van de uitkomst.
+// Een emailadres is valide wanneer:
+// * Er een @ in voorkomt
+// * Er géén , in voorkomt
+// * Er géén . in voorkomt als allerlaatste karakter (dus hotmail.com is valide, net als outlook.nl, maar outlooknl. niet)
+// ---- Verwachte uitkomsten:
+// checkEmailValidity("n.eeken@novi.nl") geeft true - want @ en punt op de juiste plek
+// checkEmailValidity("tessmellink@novi.nl") geeft true - want @ en punt op de juiste plek
+// checkEmailValidity("n.eekenanovi.nl") geeft false - want geen @
+// checkEmailValidity("n.eeken@novinl.") geeft false - want de punt mag niet als laatst
+// checkEmailValidity("tessmellink@novi,nl") geeft false - want er staat een komma in
+
+
+function checkEmailValidity (input){
+    if (input.includes(",")){
+        return false
+    } else if (input.charAt(input.length - 1) === ".") {
+        return false
+    } else if (input.includes("@", ".")){
+        return true
+    } else {
```

Comment:
Lekker bezig! Wist je dat deze nog korter kon? De hele if statement met else en de return kan dan weg. Dat ziet er dan zo uit:
`function checkEmailValidity(emaildress){
return !!(emaildress.includes("@") && !emailadress.includes(",") && emailadress.chartAt(emailadress.length - 1) !== "."); `

Het ziet er misschien een beetje gek uit! De eerste twee dubbele uitroeptekens staat eigenlijk voor een soort van mini if statement. Deze return is namelijk ook een boolean. Vervolgens gebruiken we bij de comma controle een `!` omdat we dit als logica operator kunnen gebruiken. Daarmee zeggen we dan eigenlijk hetzelfde als wat jij schrijft in je stuk code, "als dit erin zit, is het false!"